### PR TITLE
llvm: Fix for Linux

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Fixes:
```
/tmp/llvm-20210419-4659-nrx5na/llvm-project-12.0.0.src/openmp/libomptarget/include/dlwrap.h:113:36: error: specialization of 'template<long unsigned int S> struct dlwrap::type::count' in different namespace [-fpermissive]
   template <> struct dlwrap::type::count<__LINE__> {  
```

File: Formula/llvm.rb [hunk 1]
```diff
@@ -40,10 +40,10 @@ class Llvm < Formula
      (Formula["glibc"].any_version_installed? || OS::Linux::Glibc.system_version < Formula["glibc"].version)
     depends_on "glibc"
   end
+  depends_on "libffi"
   depends_on "python@3.9"
 
   uses_from_macos "libedit"
-  uses_from_macos "libffi", since: :catalina
```
Comment (Bo98) [inline Formula/llvm.rb:46]: What's the reason for this?

Comment (iMichka) [reply Formula/llvm.rb]: That was a mistake, I changed it back.

**Comment by iMichka** [comment]
```
2021-04-20T18:48:34.8873777Z [34m==>[0m [1mbrew linkage --test llvm[0m
2021-04-20T18:48:35.9749004Z [31m==>[0m [1m[31mFAILED[0m[0m
2021-04-20T18:48:35.9749619Z Missing libraries:
2021-04-20T18:48:35.9750273Z   unexpected (libc++abi.so.1)
2021-04-20T18:48:35.9750942Z Unwanted system libraries:
2021-04-20T18:48:35.9751888Z   /usr/lib/x86_64-linux-gnu/libatomic.so.1
```

**Pull request closed without merging**